mkfs/mkfs.c

```diff
@@ -17,13 +17,36 @@
 #include <errno.h>
 #include <math.h>
 #include <locale.h>
+#include <time.h>
 
 #include "exfat_ondisk.h"
 #include "libexfat.h"
 #include "mkfs.h"
 
 struct exfat_mkfs_info finfo;
 
+/* random serial generator based on current date & time */
+static unsigned int get_new_serial(void)
+{
+	struct tm *tm;
+	struct timespec ts;
+	time_t now;
+	unsigned int x;
+	unsigned int nsec_u, nsec_l;
+
+	clock_gettime(CLOCK_REALTIME, &ts);
+	now = ts.tv_sec;
+	tm = localtime(&now);
```

Comment:
minor nit, no need to check whether tm is valid or not ?

Reply:
AFAIK, there's no error defined for localtime(). 
Exfatprogs is not multi-threaded, and even if tm value is not stabilized, I think we don't need to care about the accuracy of the tm value.

Reply:
Okay!